key.go

```diff
@@ -212,20 +212,17 @@ func (k Key) ChildString(s string) Key {
 //   NewKey("/Comedy").IsAncestorOf("/Comedy/MontyPython")
 //   true
 func (k Key) IsAncestorOf(other Key) bool {
-	if other.string == k.string {
-		return false
-	}
-	return strings.HasPrefix(other.string, k.string)
+	// equivalent to HasPrefix(other, k.string + "/")
+	return len(other.string) > len(k.string) && // the child is longer
+		other.string[len(k.string)] == '/' && // the child has a separator in the right place
```

Comment:
The problem here is when `k.string == "/"` since `other.string[1]` should never be `/`.

I'll give this a try.

Reply:
any thoughts on what I can assume the state of a `Key` is? For example, do I need to run `Clean` to make sure that root isn't represented as `/./././././././`?

Reply:
Keys are cleaned when they're created. You can assume they're clean.